### DISCUSSION TITLE
Authentication - Clarification on private config

### DISCUSSION THREAD

**Discussion by cjmorrison** [koltyakov/sp-rest-proxy#132] [Q&A] [answered]
[note  I have also posted this on [Stack Overflow](https://stackoverflow.com/questions/69557498/authentication-with-sp-rest-proxy-node-sp-auth)  and will copy the answer over there for posterity should I get one]

The short of it is I am getting 403 errors after switching to "Addin only permissions" using a token/secret provided by my admins.
I believe the issue may be my configs. and I am looking for some clarification on how these are done.
```
const RestProxy = require('sp-rest-proxy');

const settings = {
 configPath: './config/private.json',
 port: 8081,
};

const restProxy = new RestProxy(settings);
restProxy.serve();
```
and private (not the actual values I am using expect for "strategy" )
```
{
  "siteUrl": "https://ORGANIZTION.sharepoint.com",
  "strategy": "OnlineAddinOnly",
  "clientId": "0000000-000000-000000-0000-00000000",
  "clientSecret": "000000000000000000000000000000",
  "realm": "00000-0000-0000-0000-000000"
}
```
I couldn’t find much on the “strategy” value on the sp-rest-proxy or the node-sp-auth side of the documentation. I can assume its OnlineAddinOnly but I’m not able to find the specific syntax for what possible values this attribute expects. I also noticed that the “clientSecret” is changing once I run the server, I assume this is an intentional encryption.

**Comment by koltyakov** [comment, marked as answer]
Please check this article https://github.com/s-KaiNet/node-sp-auth/wiki/SharePoint-Online-addin-only-authentication

403 means you’ve been authenticated but not authorized. The add-in registration has been granted no permissions or an API/resource called only supports a user context.

Regarding strategy name, it’s a helper from node-sp-auth-config library which prompts per credentials without having user a chance on a mistake, almost. The list of strategies can be seen here https://github.com/koltyakov/node-sp-auth-config/blob/master/src/config/index.ts

**Reply by cjmorrison** [reply]
Thank you. I have noticed a few differences in the XML Permissions segment that my admins have set and what is in the document you've provided. I am having my admin make the adjustments and will let know how things turn out.

**Reply by cjmorrison** [reply]
Thank you kindly for your time I am very grateful. This seems to have solved it. I am at least able to now get files and see folder data, just not file in folder data, so I'm not able to confirm that post works. though it this point I am assuming its a problem unrelated to sp-rest-proxy. (posted [here](https://stackoverflow.com/questions/69592269/sharepoint-api-getfolderbyserverrelativeurl-will-not-work-yet-getfilebyserverrel) and I will treat it as a separate issue.
